hal/src/photon/wlan_hal.cpp

```diff
@@ -845,6 +846,32 @@ int wlan_connected_rssi()
     return rssi;
 }
 
+int wlan_connected_info(void* reserved, wlan_connected_info_t* inf, void* reserved1)
+{
+    system_error_t ret = SYSTEM_ERROR_NONE;
+    int32_t rssi = 0;
+    int32_t noise = 0;
+
+    if (inf == nullptr) {
+        return SYSTEM_ERROR_INVALID_ARGUMENT;
+    }
+
+    if (wwd_wifi_get_rssi(&rssi) != WWD_SUCCESS) {
+        return SYSTEM_ERROR_UNKNOWN;
+    }
+    if (wwd_wifi_get_noise(&noise) != WWD_SUCCESS) {
+        return SYSTEM_ERROR_UNKNOWN;
+    }
+
+    inf->rssi = rssi * 100;
+    inf->snr = (rssi - noise) * 100;
+    inf->noise = noise * 100;
+
+    inf->strength = std::min(std::max(2 * (rssi + 100), 0L), 100L) * 65535 / 100;
```

Comment:
similarly here, please consider factoring out fetching the signal values from the normalization code.